datafusion/physical-plan/src/coalesce_partitions.rs

```diff
@@ -145,12 +145,14 @@ impl ExecutionPlan for CoalescePartitionsExec {
                 // least one result in an attempt to maximize
                 // parallelism.
                 let mut builder =
-                    RecordBatchReceiverStream::builder(self.schema(), input_partitions);
+                    ReceiverStream::builder(self.schema(), input_partitions);
+                let input =
+                    Arc::new(RecordBatchReceiverStreamAdaptor::new(self.input.clone()));
```

Comment:
`RecordBatchReceiverStream` was made generic, such that it could handle a buffered stream of record_batches, or the sort_orders (yielded per each merge node). Then the `RecordBatchReceiverStreamAdaptor` is used such that the record-batch-specific API (`run_input()`) was still available.

Please let me know if I should have structured this differently.